### PR TITLE
br: fix the integration tests

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

After https://github.com/pingcap/tidb/pull/30218 merged, br_other test failed due to we check lock more precisely. 
After https://github.com/pingcap/ticdc/pull/3599 cdc no longer support log_restore, so we need to remove related tests.
This PR tries to fix it.

### What is changed and how it works?
put *.sst file to backup directory & remote br_log_restore

### Check List

Tests 

- [x] Integration test

### Release note

```release-note
NONE
```

### CODE REVIEW

File: br/tests/br_other/run.sh [hunk 1]
```diff
@@ -95,6 +95,8 @@ run_curl https://$PD_ADDR/pd/api/v1/config/schedule | jq '."max-merge-region-siz
 run_curl https://$PD_ADDR/pd/api/v1/config/schedule | jq '."max-merge-region-keys"' | grep -E "^0$"
 
 backup_fail=0
+# generate 1.sst to make another backup failed.
+touch /$TEST_DIR/$DB/lock/1.sst
```
Comment (kennytm) [inline br/tests/br_other/run.sh:99]:
```suggestion
touch "$TEST_DIR/$DB/lock/1.sst"
```

**Review by kennytm — APPROVED** [review]
rest lgtm

**Review by glorv — APPROVED** [review]

**Pull request merged**